src/components/send/sendViews/initial.view.tsx

```diff
@@ -709,32 +739,36 @@ export function SendInitialView({ onNextScreen, setClaimLink, setTxHash, setChai
                     setShowGaslessAvailable(false)
                 }
                 fetchAndSetTokenPrice(tokenAddress, formwatch.chainId)
+                return () => {
+                    isCurrent = false
+                }
             }
         }
     }, [formwatch.token, formwatch.chainId, isConnected])
 
     useEffect(() => {
-        //update the chain and token when the user changes the chain in the wallet
-        if (mantleCheck) {
-            sendForm.setValue('chainId', '5000')
-            return
-        }
         if (
             currentChain &&
             currentChain?.id.toString() != formwatch.chainId &&
             !formHasBeenTouched &&
             chainDetails.some((chain) => chain.chainId == currentChain.id)
         ) {
+            // if the user is connected and the forms hasnt been touched by the user, we set the chain to the current chain and the token to native token on that chain
             sendForm.setValue(
                 'token',
                 chainDetails.find((chain) => chain.chainId == currentChain.id)?.nativeCurrency.symbol ?? ''
             )
             sendForm.setValue('chainId', currentChain.id.toString())
-        } else if (chainsToShow.length > 0 && !formHasBeenTouched) {
-            sendForm.setValue('chainId', chainsToShow[0].chainId)
-            sendForm.setValue('token', chainsToShow[0].nativeCurrency.symbol)
+        } else if (userBalances.length > 0 && !userBalances.some((balance) => balance.chainId == formwatch.chainId)) {
```

Comment:
I would personally question this entire logic of automatically switching to some chain. The problem is that the response from the mobula api arrives after a certain delay, which means that this switch also happens after a certain delay.

It has happened to me multiple times that I went to the app, saw the default of chain X (let's say Ethereum) being selected, I quickly change it to the chain that I need (let's say Arbitrum), but then automatically switches back to Ethereum, which is suuuuuuper confusing

Reply:
Fair point, will make new entry in the notion doc and address this in the next pr